addons/mail/models/mail_thread.py

```diff
@@ -3478,13 +3478,18 @@ def _notify_by_email_prepare_rendering_context(self, message, msg_vals=False,
         # user_ids if they are the author will give a sudo user, improving access performances and cache usage.
         signature = ''
         email_add_signature = msg_vals.get('email_add_signature') if msg_vals and 'email_add_signature' in msg_vals else message.email_add_signature
-        if email_add_signature:
-            author = message.env['res.partner'].browse(msg_vals.get('author_id')) if 'author_id' in msg_vals else message.author_id
-            author_user = self.env.user if self.env.user.partner_id == author else author.user_ids[0] if author and author.user_ids else False
-            if author_user:
+        email_add_footer = True
+
+        author = message.env['res.partner'].browse(msg_vals.get('author_id')) if 'author_id' in msg_vals else message.author_id
+        author_user = self.env.user if self.env.user.partner_id == author else author.user_ids[0] if author and author.user_ids else False
+
```

Comment:
And you can remove the assignation in the if-else
```suggestion
        author = message.env['res.partner'].browse(msg_vals.get('author_id')) if 'author_id' in msg_vals else message.author_id
        author_user = self.env.user if self.env.user.partner_id == author else author.user_ids[0] if author and author.user_ids else False
        email_add_footer = author_user != False
```